### PR TITLE
egl-wayland: Initialize streamImages during Pbuffer surface creation

### LINKED ISSUE #127: wayland-eglsurface.c:2562: wlEglDestroySurface: The check assertion "wl_list_empty(&surface->ctx.streamImages)" failed
I'm creating my application (c++, qt) and it doesn't work on the new egl-wayland version 1.1.15, but it works on version 1.1.13.

dmali@tony-b550gamingxv2 ~ (master) [1]> kLausPro
Path override failed for key base::DIR_APP_DICTIONARIES and path '/usr/bin/qtwebengine_dictionaries'
Path override failed for key base::DIR_APP_DICTIONARIES and path '/usr/lib/qt6/qtwebengine_dictionaries'
Path override failed for key base::DIR_APP_DICTIONARIES and path '/usr/lib/qt6/qtwebengine_dictionaries'
kLausPro: ../egl-wayland/src/wayland-eglsurface.c:2562: wlEglDestroySurface: The check assertion "wl_list_empty(&surface->ctx.streamImages)" failed.
fish: Job 1, 'kLausPro' terminated by signal SIGABRT (Abort)

git: https://github.com/dmaliog/kLausqt

### PR DESCRIPTION
**Pull request by amshafer**
Add a missing initializer for the streamImages wl_list in wlEglCreatePbufferSurfaceHook. This ensures the list is valid when we eventually destroy the surface and have to check the list contents.

This fixes crashes with qt6webengine. Fixes #127

@dkorkmazturk

### CODE REVIEW

**Comment by dkorkmazturk** [comment]
Thanks for looking into this @amshafer, LGTM.

**Comment by 1Naim** [comment]
I can confirm this fixes https://github.com/NVIDIA/egl-wayland/issues/127 on a RTX 3050 Laptop.

File: src/wayland-eglsurface.c [hunk 1]
```diff
@@ -2895,6 +2895,7 @@ EGLSurface wlEglCreatePbufferSurfaceHook(EGLDisplay dpy,
     surface->refCount = 1;
     surface->isDestroyed = EGL_FALSE;
     wl_list_init(&surface->oldCtxList);
+    wl_list_init(&surface->ctx.streamImages);
```
Comment (cubanismo) [inline src/wayland-eglsurface.c:2898]: This seems like a correct addition, but are there other fields that are missing initialization here? E.g., surface->ctx.streamImageMutex?

Would it be a better fix to have a common surface initializer helper function that covers all the surface-type-agnostic intialization of a surface structure?

Comment (amshafer) [reply src/wayland-eglsurface.c]: Thanks, added this.

**Review by cubanismo — APPROVED** [review]

**Pull request merged**